modules/kubernetes-addons/opentelemetry-operator/main.tf

```diff
@@ -1,17 +1,33 @@
-module "operator" {
-  source        = "../helm-addon"
-  helm_config   = local.helm_config
-  irsa_config   = null
+module "cert_manager" {
+  source        = "../cert-manager"
   addon_context = var.addon_context
-
-  depends_on = [kubernetes_namespace_v1.prometheus]
 }
 
-resource "kubernetes_namespace_v1" "prometheus" {
+resource "kubernetes_namespace_v1" "adot" {
   metadata {
     name = local.helm_config["namespace"]
```

Comment:
we have to be careful with this - if the EKS addon for ADOT expects a [specific namespace](https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/627/files#diff-0393764fcbfc5379a2df275af001d20d3e8c190ab4570efcb83340c3eeee8bc1R5) that is not configurable, then we have to match that. Do checkout the CoreDNS addon where we have two distinct set of values - one for EKS addon, one for self-managed addon. This ensures that if the users selects the self-managed route, we can honor that flexibility, but if they select the EKS addon, we have to match what the EKS addon expects